packages/navigation/src/HeaderButton.js

```diff
@@ -0,0 +1,164 @@
+// @flow
+
+import * as React from 'react';
+import { Platform, View } from 'react-native';
+import {
+  Touchable,
+  Color,
+  StyleSheet,
+  Icon,
+  Text,
+  type StylePropType,
+} from '@kiwicom/mobile-shared';
+import {
+  type TranslationKeysType,
+  type TranslationType,
+  Translation,
+} from '@kiwicom/mobile-localization';
+
+type Props = {|
+  +onPress: () => void,
+  +onLongPress?: (React.ElementRef<typeof Touchable>) => void,
+  +children:
+    | React.Element<typeof HeaderButton.Text>
+    | React.Element<typeof Icon>,
+  +left?: boolean,
+  +style?: StylePropType,
+  +disabled?: boolean,
+|};
+
+type State = {|
+  reference: React.ElementRef<typeof Touchable> | null,
+|};
+
+type CloseModalProps = {|
+  +onPress: () => void,
+  +textId?: TranslationKeysType,
+|};
+
+export default class HeaderButton extends React.PureComponent<Props, State> {
+  static Right = (props: Props) => (
+    <HeaderButton {...props} style={[props.style, styles.right]} />
+  );
+
+  static Left = (props: Props) => (
+    <HeaderButton {...props} style={[props.style, styles.left]} />
+  );
+
+  static Text = ({
+    children,
+    disabled,
+  }: {|
+    +children: TranslationType,
+    +disabled?: boolean,
+  |}) => (
+    <Text
+      style={[styles.headerButtonText, disabled ? styles.disabledButton : null]}
+    >
+      {React.cloneElement(children, {
+        textTransform: Platform.OS === 'android' ? 'uppercase' : '',
+      })}
+    </Text>
+  );
+
+  static CloseModal = ({ onPress, textId }: CloseModalProps) => (
+    <HeaderButton.Left onPress={onPress}>
+      {Platform.OS === 'android' ? (
+        <Icon name="close" size={24} color={Color.brand} />
+      ) : (
+        <HeaderButton.Text>
+          <Translation id={textId || 'shared.button.close'} />
```

Comment:
I think it is better to pass translations than the id, so 

```
{textId || <Translation id="shared.button.close" />
```

Reply:
I am not sure I understand this one, `{textId || <Translation id="shared.button.close" />` this won't work because `textId` is something like "hotels.button.cancel` so it needs to be used with `Translation` right?

Reply:
Right, I am sorry, I explained my self badly

```
// If close modal is like:
static CloseModal = ({ onPress, text }: {|+onPress: () => void, +text?: TranslationType |}) {
if (text === undefined) return <Translation id="shared.button.close" />
return text;
}

// Then I can use it like
<CloseModal onPress={() => {}} text={<Translation id="mmb.lol.lol" />} />

```